src/pages/instances/InstanceList.tsx

```diff
@@ -354,61 +451,24 @@ const InstanceList: FC = () => {
               </>
             )}
             {hasInstances && selectedNames.length === 0 && (
-              <>
-                <SearchBox
-                  className="search-box margin-right u-no-margin--bottom"
-                  name="search-instance"
-                  type="text"
-                  onChange={(value) => {
-                    setQuery(value);
-                  }}
-                  placeholder="Search"
-                  value={query}
-                  aria-label="Search"
+              <div className="search-wrapper margin-right u-no-margin--bottom">
+                <SearchAndFilter
```

Comment:
Maybe contain this and all the code to build it in a dedicated component? When combining the filters into a single state, the interface (as amount of props) should be fairly small.

Reply:
Good idea, I was already planning to do this, because the `InstanceList` is already super-polluted - this is to say that, as a future improvement, we should extract many more things out of it - but eventually I got distracted by the countless layout issues created by the introduction of the `SearchAndFilter` component...